python-backend/main.py

```diff
@@ -124,8 +124,9 @@ async def data_socket(websocket: WebSocket):
         except WebSocketDisconnect:
             logger.debug("Websocket disconnected")
             break
-        except WebSocketException:
-            logger.debug("Websocket exception")
+        except WebSocketException as e:
+            logger.debug(f"Websocket exception {e}")
+            await websocket.close()
```

Comment:
This one probably isn't necessary now that the call to `close` on 132 is outside of the `while True` loop.